### PR TITLE
Feature/is58/stats and energies

### PR DESCRIPTION
**Pull request by BennyKayer**
Both done got little ahead of myself and changed navigation too, sry for not creating a fix branch.

### CODE REVIEW

**Comment by BennyKayer** [comment]
onClicks on stats and energies icons are temporary.

File: src/components/navigation/navigation.component.jsx [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
-    }
-    componentWillUnmount() {
-        window.removeEventListener("resize", this.handleWindowSizeChange);
-    }
-    handleWindowSizeChange() {
-        this.setState({ width: window.innerWidth });
-    }
-    render() {
-        const isMobile = this.state.width < 768;
+const Navigation = ({ width, setWidth }) => {
```
Comment (kulaszewicz) [inline src/components/navigation/navigation.component.jsx]: I don't think that using Redux just to handle screen width and use that information just in one place is a good idea. Using Redux doesn't mean that all the state connected features has to be solved by Redux, It is actually a good practice to find balance and know to when use Redux store and when to use react state. Here when you use that width property in one place there's no point in storing that in global store. Besides on Devtools when resizing redux logger goes crazy :p.

File: src/components/navigation/navigation.component.jsx [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
+    window.addEventListener("resize", value => setWidth(value));
```
Comment (kulaszewicz) [inline src/components/navigation/navigation.component.jsx]: Has to be removed on component unmount

File: src/components/progress-bar/progress-bar.component.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const ProgressBar = ({ children, barType, stat }) => {
+    const barStyle = {
+        width: `${stat}%`,
+    };
+    return (
+        <div className="container">
+            <div className="progress">
+                {children}
+                <div className={`bar ${barType}`} style={barStyle}></div>
```
Comment (kulaszewicz) [inline src/components/progress-bar/progress-bar.component.jsx]: <div className={`bar ${barType}`} style={barStyle} /> is enough, there's no content inside

File: src/components/navigation/navigation.component.jsx (hunk 1 above)
Comment (BennyKayer) [reply src/components/navigation/navigation.component.jsx]: True logger goes bananas, width isn't going to be used elsewhere. I've just gone on the redux-spree :p

File: src/components/navigation/navigation.component.jsx [hunk 4]
```diff
@@ -3,29 +3,23 @@ import React from "react";
...
+            setWidth(window.innerWidth);
+        };
+        window.addEventListener("resize", handleResize);
+
+        return () => {
+            window.removeEventListener("resize", handleResize);
         };
-        this.handleWindowSizeChange = this.handleWindowSizeChange.bind(this);
-    }
-    UNSAFE_componentWillMount() {
```
Comment (kulaszewicz) [inline src/components/navigation/navigation.component.jsx]: Is it necessary to use when already set in constructor?

**Pull request merged**